client/trino-jdbc/src/main/java/io/trino/jdbc/NonRegisteringTrinoDriver.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.jdbc;
+
+import okhttp3.OkHttpClient;
+
+import java.io.Closeable;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverPropertyInfo;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+import static io.trino.client.OkHttpUtil.setupChannelSocket;
+import static io.trino.client.OkHttpUtil.userAgent;
+import static io.trino.jdbc.DriverInfo.DRIVER_NAME;
+import static io.trino.jdbc.DriverInfo.DRIVER_VERSION;
+import static io.trino.jdbc.DriverInfo.DRIVER_VERSION_MAJOR;
+import static io.trino.jdbc.DriverInfo.DRIVER_VERSION_MINOR;
+
+public class NonRegisteringTrinoDriver
```

Comment:
TrinoDriverBase

Reply:
This is the full driver, with the only difference being that it doesn't register. I can explain more offline.